akka-http-core/src/main/scala/akka/http/scaladsl/ClientTransportWithCustomResolver.scala

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.http.scaladsl
+
+import java.net.InetSocketAddress
+
+import akka.actor.ActorSystem
+import akka.annotation.ApiMayChange
+import akka.http.scaladsl.Http.OutgoingConnection
+import akka.http.scaladsl.settings.ClientConnectionSettings
+import akka.stream.scaladsl.{ Flow, Tcp }
+import akka.util.ByteString
+
+import scala.concurrent.{ ExecutionContext, Future }
+
+@ApiMayChange
+trait ClientTransportWithCustomResolver extends ClientTransport {
+  override def connectTo(host: String, port: Int, settings: ClientConnectionSettings)(implicit system: ActorSystem): Flow[ByteString, ByteString, Future[Http.OutgoingConnection]] = {
+    implicit val ec = system.dispatcher
+    futureFlow(
```

Comment:
Maybe `lazyInitAsync` would work on 2.5 and 2.6? Maybe you could prepend an empty `ByteString()` to that flow, so it will be initialized immediately on materialization.

Reply:
I added a commit that does that so we don't need the forward port for now (might be slightly less efficient because of an extra indirection in the stream).